blocks/form/form.js

```diff
@@ -1,36 +1,91 @@
 import { sampleRUM } from '../../scripts/lib-franklin.js';
 
+const SITE_KEY = '6LeMTDUlAAAAAMMlCNN-CT_qNsDhGU2xQMh5XnlO';
+const FORM_SUBMIT_ENDPOINT = 'http://localhost:8787';
+
+function loadScript(url) {
+  const head = document.querySelector('head');
+  let script = head.querySelector(`script[src="${url}"]`);
+  if (!script) {
+    script = document.createElement('script');
+    script.src = url;
+    script.async = true;
+    head.append(script);
+    return script;
+  }
+  return script;
+}
+
 function constructPayload(form) {
   const payload = {};
+  const attachments = {};
   [...form.elements].forEach((fe) => {
     if (fe.type === 'checkbox') {
       if (fe.checked) payload[fe.id] = fe.value;
+    } else if (fe.type === 'file' && fe.files?.length > 0) {
+      attachments[fe.name] = fe.files;
     } else if (fe.id) {
       payload[fe.id] = fe.value;
     }
   });
-  return payload;
+  return { payload, attachments };
 }
 
-async function submitForm(form) {
-  const payload = constructPayload(form);
-  const resp = await fetch(form.dataset.action, {
+async function submissionFailure(error, form) {
+  alert(error); // TODO define error mechansim
+  form.setAttribute('data-submitting', 'false');
+  form.querySelector('button[type="submit"]').disabled = false;
+}
+
+async function submitForm(form, token) {
```

Comment:
Separate Concerns: Move the code that constructs the request (headers and body) to a separate function. This will make the submitForm function more focused on its main responsibility, which is to submit the form.